cmake/ClioPackage.cmake

```diff
@@ -0,0 +1,41 @@
+set(CPACK_PACKAGING_INSTALL_PREFIX "/opt/clio")
+include("${CMAKE_CURRENT_LIST_DIR}/ClioVersion.cmake")
+
+set(VERSION ${CLIO_VERSION})
+
+if (VERSION MATCHES "^([0-9]+)\.([0-9]+)\.([0-9]+)(-(b|rc|hf)([0-9]+))?$|^([0-9a-zA-Z]+)-(.+)-([0-9a-fA-F]+)$")
+  if (CMAKE_MATCH_2)
+    set(MAJOR "${CMAKE_MATCH_1}")
+    set(MINOR "${CMAKE_MATCH_2}")
+    set(PATCH "${CMAKE_MATCH_3}")
+    message(DEBUG " Major: ${MAJOR}")
+    message(DEBUG " Minor: ${MINOR}")
+    message(DEBUG " Patch: ${PATCH}")
+    set(release_type "release ${MAJOR}.${MINOR}.${PATCH}")
+    if (CMAKE_MATCH_4)
+      set(PRE_TYPE "${CMAKE_MATCH_5}")
+      set(PRE_NUM "${CMAKE_MATCH_6}")
+      message(DEBUG " Pre-release type: ${PRE_TYPE}")
+      message(DEBUG " Pre-release number number: ${PRE_NUM}")
+      set(release_type "pre-${release_type}-${PRE_TYPE}-${PRE_NUM}")
+    endif ()
+  else ()
+    set(DATE "${CMAKE_MATCH_7}")
+    set(BRANCH "${CMAKE_MATCH_8}")
+    set(GIT_HASH "${CMAKE_MATCH_9}")
+    message(DEBUG " Date: ${DATE}")
+    message(DEBUG " Branch: ${BRANCH}")
+    message(DEBUG " Git hash: ${GIT_HASH}")
+    set(release_type "development ${DATE}-${BRANCH}-${GIT_HASH}")
+  endif ()
+  set(CPACK_PACKAGE_VERSION "${CMAKE_MATCH_0}")
+else ()
+  message(FATAL_ERROR "Version string '${CLIO_VERSION}' did not match expected patterns.")
+endif ()
```

Comment:
This is so difficult to read and understand
Maybe setting variables inside `ClioVersion.cmake` would be a better solution.

Because now we set long name in `ClioVersion.cmake` and then parse it back here, which doesn't make sense.